### ISSUE TITLE
Make "edit" action dot-repeatable.

### ISSUE THREAD

**Issue by JeanMertz** [yetone/avante.nvim#365]
Say I have a bunch of undocumented functions.

I select the first function, and tell it to "add documentation with LuaLS type annotations".

The edit is successful (awesome, btw!) but now I want to do the same for the next 5 functions. Obviously in the same file, I can (probably) just select them all at once, but ideally, I could just `.` repeat my last edit similar to other edit actions I do in my buffer.

**Comment by oxysoft** [comment]
You mean the `<leader>ae` action right? I agree, a 'semantic action repeat' would be insane. In fact, you could even make it work when there is no selection by asking the LLM to recognize the common pattern at the caret location, and choose by itself a visual selection to refactor first before running the prompt.

**Comment by yetone** [comment]
Thank you for the suggestion, I will look into how to implement it.

**Comment by yetone** [comment]
Do you expect the functionality to require selecting the repeat code block in advance before pressing <kbd>.</kbd>?

**Comment by JeanMertz** [comment]
Yeah, although initially, I would like for me to make a selection, and then dot-repeat would trigger the last edit query and apply to diff in one go.

**Comment by aarnphm** [comment]
very initial scuffed first idea that doesn't involve state machine:
- just memoized last action and integrate with vim-repeat.

**Comment by aarnphm** [comment]
Thinking more about this, best is to have a snippet over advanced configuration for users who want to use this instead of building this into the core offering.

This can be setup via setup hooks in lazy or otherwise (autocmd).
